crates/net/src/http/query.rs

```diff
@@ -0,0 +1,104 @@
+use gloo_utils::iter::UncheckedIter;
+use js_sys::{Array, Map};
+use std::fmt;
+use wasm_bindgen::{JsCast, UnwrapThrowExt};
+
+/// A sequence of URL query parameters, wrapping [`web_sys::UrlSearchParams`].
+pub struct QueryParams {
+    raw: web_sys::UrlSearchParams,
+}
+
+impl Default for QueryParams {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[allow(dead_code)]
+impl QueryParams {
+    /// Create a new empty query parameters object.
+    pub fn new() -> Self {
+        // pretty sure this will never throw.
+        Self {
+            raw: web_sys::UrlSearchParams::new().unwrap_throw(),
+        }
+    }
+
+    /// Create [`QueryParams`] from [`web_sys::UrlSearchParams`] object.
+    pub fn from_raw(raw: web_sys::UrlSearchParams) -> Self {
+        Self { raw }
+    }
+
+    /// Append a parameter to the query string.
+    pub fn append(&self, name: &str, value: &str) {
+        self.raw.append(name, value)
+    }
+
+    /// Get the value of a parameter. If the parameter has multiple occurrences, the first value is
+    /// returned.
+    pub fn get(&self, name: &str) -> Option<String> {
+        self.raw.get(name)
+    }
+
+    /// Get all associated values of a parameter.
+    pub fn get_all(&self, name: &str) -> Vec<String> {
+        self.raw
+            .get_all(name)
+            .iter()
+            .map(|jsval| jsval.as_string().unwrap_throw())
+            .collect()
+    }
+
+    /// Remove all occurrences of a parameter from the query string.
+    pub fn delete(&self, name: &str) {
+        self.raw.delete(name)
+    }
+
+    /// Iterate over (name, value) pairs of the query parameters.
+    pub fn entries(&self) -> impl Iterator<Item = (String, String)> {
```

Comment:
It should be `iter` but that's not blocking the PR. I'll rename it later if you don't want to

Reply:
Changed. Thanks for reviewing 👍🏻